mobile/src/libs/components/input-file/input-file.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import DocumentPicker from 'react-native-document-picker';
+
+import { FIRST_ARRAY_INDEX } from '#libs/constants/constants';
+import { AppColor } from '#libs/enums/enums';
+import { requestMediaPermissions } from '#libs/helpers/helpers';
+import { useAppDispatch, useFormController } from '#libs/hooks/hooks';
+import {
+  type FormControl,
+  type FormFieldErrors,
+  type FormFieldPath,
+  type FormFieldValues,
+} from '#libs/types/types';
+import { appActions } from '#slices/app/notifications';
+
+import { Icon, Pressable, Text, View } from '../components';
+import { styles } from './styles';
+
+type Properties<T extends FormFieldValues> = {
+  control: FormControl<T, null>;
+  errors: FormFieldErrors<T>;
+  name: FormFieldPath<T>;
+  label: string;
+};
+
+const InputFile = <T extends FormFieldValues>({
+  control,
+  name,
+  errors,
+  label,
+}: Properties<T>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+
+  const { value, onChange } = field;
+  const dispatch = useAppDispatch();
+
+  const error = errors[name]?.message;
+  const fileData = value?.data as File;
+  const hasError = Boolean(error);
+  const hasValue = Boolean(fileData);
+
+  const handlePickFile = async (): Promise<void> => {
+    try {
+      const hasPermissions = await requestMediaPermissions();
+      if (hasPermissions) {
+        const result = await DocumentPicker.pick({
+          type: [DocumentPicker.types.audio],
+        });
+        const file = result[FIRST_ARRAY_INDEX];
+        onChange({ data: file, type: file?.type, size: file?.size });
```

Comment:
What if the user did not select anything? I think `file` will be `undefined` in that case

Reply:
If the user doesn't select anything, then the code in the catch block is executed